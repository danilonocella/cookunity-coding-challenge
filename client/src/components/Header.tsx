import { FC } from "react";
import { AppBar, Box, Button, Toolbar, Typography } from "@mui/material";
import CatchingPokemonIcon from "@mui/icons-material/CatchingPokemon";
import { Link } from "react-router-dom";

const navItems = [
  { label: "Home", url: "/" },
  { label: "About", url: "/about" },
];

const Header: FC = () => (
  <AppBar position="static" data-testid="header">
    <Toolbar
      sx={{
        backgroundColor: (theme) => theme.palette.secondary.main,
      }}
    >
      <Typography
        color="secondary"
        variant="h6"
        component="div"
        sx={{ flexGrow: 1, display: { xs: "none", sm: "block" } }}
      >
        <CatchingPokemonIcon sx={{ color: "#FFF", transform: 'scaleY(-1)' }} />
      </Typography>
      <Box sx={{ display: { xs: "none", sm: "block" } }}>
        {navItems.map((item) => (
          <Link key={item.label} to={item.url}>
            <Button sx={{ color: "#FFF" }}>{item.label}</Button>
          </Link>
        ))}
      </Box>
    </Toolbar>
  </AppBar>
);

export default Header;
