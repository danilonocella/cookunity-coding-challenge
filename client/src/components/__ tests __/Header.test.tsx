import { render } from "@testing-library/react";
import Header from "../Header";
import { BrowserRouter } from 'react-router-dom';

test("Header component matches snapshot", () => {
  const { container } = render(
    <BrowserRouter>
      <Header />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
