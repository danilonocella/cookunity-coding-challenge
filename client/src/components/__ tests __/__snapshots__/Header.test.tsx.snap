// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    data-testid="header"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1bdtkqf-MuiToolbar-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h6 css-1jggtah-MuiTypography-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-9vjig-MuiSvgIcon-root"
          data-testid="CatchingPokemonIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.5 12c0 1.38-1.12 2.5-2.5 2.5S9.5 13.38 9.5 12s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5m7.5 0c0 5.52-4.48 10-10 10S2 17.52 2 12 6.48 2 12 2s10 4.48 10 10m-2 0h-4c0-2.21-1.79-4-4-4s-4 1.79-4 4H4c0 4.41 3.59 8 8 8s8-3.59 8-8"
          />
        </svg>
      </div>
      <div
        class="MuiBox-root css-130f8nx"
      >
        <a
          href="/"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1akgezq-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Home
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
        <a
          href="/about"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1akgezq-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            About
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
      </div>
    </div>
  </header>
</div>
`;
